main/ble2mqtt.c

```diff
@@ -156,27 +157,51 @@ static void cleanup(void)
     ota_unsubscribe();
 }
 
-/* Wi-Fi callback functions */
-static void wifi_on_connected(void)
+/* Network callback functions */
+static void network_on_connected(void)
 {
-    log_start(config_log_host_get(), config_log_port_get());
-    ESP_LOGI(TAG, "Connected to WiFi, connecting to MQTT");
     mqtt_connect(config_mqtt_host_get(), config_mqtt_port_get(),
         config_mqtt_client_id_get(), config_mqtt_username_get(),
         config_mqtt_password_get(), config_mqtt_ssl_get(),
         config_mqtt_server_cert_get(), config_mqtt_client_cert_get(),
         config_mqtt_client_key_get());
 }
 
-static void wifi_on_disconnected(void)
+static void wifi_on_connected(void)
+{
+    log_start(config_log_host_get(), config_log_port_get());
+    ESP_LOGI(TAG, "Connected to WiFi, connecting to MQTT");
+    network_on_connected();
+}
+
+static void eth_on_connected(void)
+{
+    log_start(config_log_host_get(), config_log_port_get());
+    ESP_LOGI(TAG, "Connected to Ethernet, connecting to MQTT");
+    network_on_connected();
+}
+
+static void network_on_disconnected(void)
```

Comment:
I see no reason to separate to the two implementations, especially since the only difference is the log message.
I suggest renaming the `wifi_on_xxx` functions to `network_on_xxx` as you already did and simply modify the log messages to "Connected to the network...".
This goes for `wifi_on_disconnected`, and the events below.